_posts/2021-04-21-frequency-band-control-in-unetstack.md

```diff
@@ -0,0 +1,199 @@
+
+The following will send out a 1s BPSK signal with constellation (-1, 1) at the highest rate possible, i.e., every alternating complex sample is (+1,0) followed by (-1,0). 
+
+```
+> s = [1,0, -1, 0]*6000
+[1, 0, -1, 0, 1, 0, -1, 0, 1, 0, <<snip>> -1, 0, 1, 0, -1, 0, 1, 0, -1, 0]
+
+> phy << new TxBasebandSignalReq(signal: s)
+
+```
+
+Passband PSD plot from the diagnostics scope showing the main frequencies at 6kHz and 18kHz implying full use of the band 6 to 18kHz.
+<p align="center"><img src="../assets/img/freqBandControl/bpsk-high.png" style="zoom:50%;"/></p>
+
+
+The following will send out a 1s BPSK signal at a half the highest rate possible. 
+
+```
+> s = [1,0, 1, 0, -1, 0, -1, 0]*3000
+[1, 0, -1, 0, 1, 0, -1, 0, 1, 0, <<snip>> -1, 0, 1, 0, -1, 0, 1, 0, -1, 0]
+
+> phy << new TxBasebandSignalReq(signal: s)
+
+```
+
+Passband PSD plot from the diagnostics scope showing the main frequencies at 9kHz and 15kHz implying use of 6kHz bandwidth
+<p align="center"><img src="../assets/img/freqBandControl/bpsk-low.png" style="zoom:50%;"/></p>
+
+Note that in the strict sense, unless a band limiting RRC (Raised Root Cosine) filter or equivalent is used, the effective bandwidth does not reduce correctly for BPSK with random binary data. The proper implementation of BPSK or general PSK is not aimed in this blog.
+
+The above is intended to give the reader a flavour of how frequency band used changes with signals. The reader no doubt will be interested in testing different types of modulation schemes, and using the PSD scope, the frequency band used can be visualized. Surely it can also be estimated from communication theory.
+
+## OFDM frequency band control
+
+OFDM is one of the main modulation schemes provided with Subnero modems. Controlling the bandwidth and centre of the frequency band used is very simple for this scheme.
+
+OFDM uses a number of carriers to modulate the baseband data and the number is specified via `phy[2].nc`. These carriers span the total available bandwidth (24kHz for Medium Frequency Subnero modem variants), i.e., from 12kHz to 36kHz. 
+
+The parameter `phy[2].bw`. controls a reduction in the usage of this total bandwidth. I.e., if we specifiy `phy[2].bw=0.5`, it reduces the bandwidth used by half, i.e, 24kHz/2 = 12kHz. 
+
+The other parameter involved in bandwidth control is `phy[2].dc0`. This controls the starting carrier number. 
+
+`dc0 >= -nc/2 || dc0 <= nc/2`
+
+This can be used to shift the centre of the frequency band "left" or "right". 
+
+As mentioned above, to set the band to match the transducer, we need `C=26kHz` and `B=12kHz`.
+
+The default modem parameters are set as follows to achieve this.
+```
+> phy[2]
+  ...
+  bw = 0.5
+  dc0 = 85
+  nc = 1024
+```
+
+`phy.dc0=85` shifts the centre by approximately `2000Hz`. Carriers go from -512 to +512 to span the band 12kHz to 36kHz. `85/512 ~= 1/6`, so the centre will shift roughly by `1/6` of 12kHz = 2000Hz (the positive half band) to 26kHz from 24kHz.
```

Comment:
Good practice to have spaces around `=` and before units. Also no need to have numbers or units typeset as code!

Reply:
Applies in many places, won't comment everywhere, but it's understood that the fix is needed everywhere